client/securedrop_client/sdk/__init__.py

```diff
@@ -147,12 +164,102 @@ def _send_json_request(
         if timeout:
             data["timeout"] = timeout
 
+        env = {}
+        if self.development_mode:
+            env["SD_PROXY_ORIGIN"] = self.server
+
+        # Streaming
+        if stream:
+            fobj = tempfile.TemporaryFile("w+b")
+
+            content_length = 0
+            filename = ""
+            sha256sum = ""
+
+            retry = 0
+            bytes_written = 0
+            download_finished = False
+
+            while not download_finished and retry < DOWNLOAD_RETRY_LIMIT:
+                try:
+                    # Open the process
+                    proc = subprocess.Popen(
+                        self._rpc_target(),
+                        stdin=subprocess.PIPE,
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.PIPE,
+                        env=env,
+                    )
+
+                    # Send the data, included the range if we are retrying
+                    if retry > 0:
+                        data["headers"]["Range"] = f"bytes={bytes_written}-"
+
+                    data_str = json.dumps(data).encode()
+                    if proc.stdin is not None:
+                        proc.stdin.write(data_str)
+                        proc.stdin.close()
+
+                    # Save the response headers from the first request
```

Comment:
the headers are only output to stderr after the whole thing has finished streaming, so I don't think it can be fetched from the first request if it is interrupted. (we might also want to have the server emit a `full-content-length` or something header to indicate what the full length of the file is, not just what is being transmitted in this request??)